### ISSUE TITLE
Error: Accessories must be created with a non-empty displayName

### ISSUE THREAD

**Issue by erngyver** [nnance/homebridge-august-locks#1]
**Describe The Bug:**
Thank you for building this!

Installed and configured plugin with correct credentials, 3 august devices on account. Observing error message in logs and no locks configured in homebridge UI or Apple Home app

**To Reproduce:**
* install Plugin
* Configure plugin
* Restart Homebridge
* observe error

**Expected behavior:**
Smart locks on august account should be configured in homebridge and home app, no error message in logs

**Logs:**

```
[12/16/2021, 11:38:57 AM] [August Locks] statusCode: 200
[12/16/2021, 11:38:57 AM] [August Locks] statusCode: 401
[12/16/2021, 11:38:57 AM] [August Locks] Adding new accessory: undefined
(node:57485) UnhandledPromiseRejectionWarning: AssertionError [ERR_ASSERTION]: Accessories must be created with a non-empty displayName.
    at new Accessory (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Accessory.ts:417:11)
    at new PlatformAccessory (/usr/local/lib/node_modules/homebridge/src/platformAccessory.ts:70:9)
    at AugustSmartLockPlatform.discoverDevices (/usr/local/lib/node_modules/homebridge-august-locks/src/platform.ts:108:27)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:57485) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:57485) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[12/16/2021, 11:38:57 AM] [August Locks] statusCode: 200

```

**Plugin Config:**

```json
{
    "bridge": {
        "name": "Homebridge 925D",
        "username": "0E:29:5B:B5:92:5D",
        "port": 51219,
        "pin": "{{redacted}}"
    },
    "accessories": [],
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "auth": "form",
            "theme": "auto",
            "tempUnits": "f",
            "lang": "auto",
            "platform": "config"
        },
        {
            "refreshToken": "{{redacted}}",
            "hideDoorbellSwitch": true,
            "hideCameraSirenSwitch": true,
            "hideAlarmSirenSwitch": true,
            "_bridge": {
                "username": "{{redacted}}",
                "port": 45650
            },
            "platform": "Ring"
        },
        {
            "name": "orbit",
            "email": "{{redacted}}",
            "password": "{{redacted}}",
            "platform": "orbit"
        },
        {
            "platform": "Ecobee 3 Sensors",
            "name": "Ecobee",
            "exclude_sensors": false,
            "exclude_humidity_sensors": true,
            "exclude_occupancy_sensors": true,
            "exclude_temperature_sensors": false,
            "exclude_thermostat": false
        },
        {
            "name": "August Locks",
            "email": "{{redacted}}",
            "password": "{{redacted}}",
            "installId": "{{redacted}}",
            "platform": "AugustLocks"
        }
    ],
    "disabledPlugins": [
        "homebridge-ecobee3-sensors"
    ]
}
```

**Screenshots:**

**Environment:**

* **Plugin Version**: 1.0.0
* **Homebridge Version**: 1.3.8
* **Node.js Version**: v14.17.0
* **NPM Version**: 6.14.13
* **Operating System**: macOS 12.0.1 (Macbook Air 13-inch Early 2015)

**Comment by mojodean** [comment]
Was able to reproduce issue following same pattern.

**Comment by nnance** [comment]
I believe this is related to a couple of bugs in how the plugin was handling auth conditions.    @erngyver @mojodean can you confirm if version 1.3.0 resolves your issue

**Comment by mojodean** [comment]
@nnance can confirm, 1.3.0 resolved the auth issue for me. Upon further testing, it looks like the plugin is having issues identifying the locks as locks and wasn't providing any functionality, but was exposing them as devices in HomeKit. I'll check this out again later and open a new issue if it seems to still be the case then.

**Comment by erngyver** [comment]
@nnance - confirmed this is resolved as of 1.3.0 for me
